### PR TITLE
feat: Automatically link urls found in check and notification status messages

### LINKED ISSUE #1420: Automatically link urls found in check and notification status messages
It would be handy to automatically link urls found in status messages, similar to the way we do it in table graphs today. Clicking that link would open it in a new tab.

image

### PR DESCRIPTION
**Pull request by joelfried**
Closes #1420 

Automatically adds links to the TableRow component (used by both the checks and notifications status pages) so that any link in a TableRow gets built into a link identically to the way it's done in EventTable.

Looks like this in action:
Checks-link

### CODE REVIEW

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 54 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 100 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=1598)

File: src/visualization/types/Table/TableCell.tsx [hunk 1]
```diff
@@ -44,7 +44,7 @@ const URL_REGEXP = /((http|https)?:\/\/[^\s]+)/g
 
 // NOTE: rip this out if you spend time any here as per:
 // https://stackoverflow.com/questions/1500260/detect-urls-in-text-with-javascript/1500501#1500501
-function asLink(str) {
+export function asLink(str) {
```
Comment (Asalem1) [inline src/visualization/types/Table/TableCell.tsx:47]: Since this is an exported function, should we add a unit test here to validate this?

**Review by Asalem1 — CHANGES_REQUESTED** [review]

File: src/visualization/types/Table/TableCell.tsx (hunk 1 above)
Comment (Asalem1) [reply src/visualization/types/Table/TableCell.tsx:47]: Like:

```
expect(asLink('string').toEqual('string')
expect(asLink('http://www.google.com').toEqual(something else that isn't the string)
```

File: src/eventViewer/components/TableRow.tsx [hunk 2]
```diff
@@ -22,7 +23,7 @@ const TableRow: FC<Props> = ({row, style, fields}) => {
           } else if (Component) {
             content = <Component key={rowKey} row={row} />
           } else {
-            content = String(row[rowKey])
+            content = asLink(row[rowKey])
```
Comment (subirjolly) [inline src/eventViewer/components/TableRow.tsx:26]: Just making sure that it's intended for this content to be an array too?

File: src/visualization/types/Table/TableCell.tsx (hunk 1 above)
Comment (subirjolly) [reply src/visualization/types/Table/TableCell.tsx:47]: 👍   to unit tests

File: src/eventViewer/components/TableRow.tsx (hunk 2 above)
Comment (joelfried) [reply src/eventViewer/components/TableRow.tsx:26]: The issue wanted the data in this component handled the same as in the other case so I believe that is appropriate.  It works as I'd expect it to.

File: src/visualization/types/Table/TableCell.tsx (hunk 1 above)
Comment (joelfried) [reply src/visualization/types/Table/TableCell.tsx:47]: Tests added.  And they caught an edge case so great call guys!  If there's any cases you think need further addition let me know.

File: src/visualization/types/Table/TableCell.test.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
+import {asLink} from './TableCell'
+
+describe('TableCell.asLink', () => {
+  it('Properly handles various base data types', () => {
+    expect(asLink('a regular string')).toEqual('a regular string')
+    expect(asLink(NaN)).toEqual(NaN)
+    expect(asLink(true)).toEqual(true)
+    expect(asLink(42)).toEqual(42)
+    expect(asLink(undefined)).toEqual(undefined)
+    expect(asLink(null)).toEqual(null)
```
Comment (Asalem1) [inline src/visualization/types/Table/TableCell.test.ts:10]: can we get an `array`, `object`, and `json object` to test here

**Review by Asalem1 — APPROVED** [review]
Better, can we add some tests to more data types and also check to see if the following link will be linkable?

`https://cloud2.influxdata.com/signup`

File: src/visualization/types/Table/TableCell.test.ts (hunk 3 above)
Comment (joelfried) [reply src/visualization/types/Table/TableCell.test.ts:10]: Added per our conversation

**Review by subirjolly — APPROVED** [review]
